src/reducers/product.js

```diff
@@ -0,0 +1,91 @@
+import {
+  CHANGE_QTY_PRODUCT,
+  CHANGE_QTY_PRODUCT_DOWN,
+  CHANGE_QTY_PRODUCT_UPP,
+  GET_PRODUCT,
+  SET_QTY_ENTER_VALUE,
+  CHANGE_SLIDE,
+  CHOOSE_IMG
+} from "../constants";
+
+const BASIC_STATE = {
+  details: [],
+  qtyProducts: "",
+  sliderStore: [],
+  slideIndex: 0,
+  slideActive: ""
+};
+export const product = (
+  state = BASIC_STATE,
+  { type, productDetails, qtyProduct, enteredValue, imgIndex }
+) => {
+  switch (type) {
+    case GET_PRODUCT:
+      let newState = { ...state };
+      newState["details"] = productDetails;
+      newState["sliderStore"] = productDetails.slider;
+      return newState;
+
+    case CHANGE_SLIDE:
+      let newStateSlide = { ...state };
```

Comment:
Reducer must be pure function

All changes must be on actions

Think about it!!!